File: cuaderno_mantenimiento_backend/src/intervention_details/dto/create-intervention_details.dto.ts
import { IsUUID, IsString, IsNotEmpty, IsOptional, IsIn } from 'class-validator';

export class CreateInterventionDetailsDto {
  @IsNotEmpty()
  @IsString()
  elemento: string;

  @IsNotEmpty()
  @IsString()
  @IsIn(['Bueno', 'Regular', 'Malo', 'Sustituido'], {
    message: 'status must be either "Bueno", "Regular", "Malo" or "Sustituido"',
  })
  estado: string;

  @IsOptional()
  @IsString()
  marca?: string;

  @IsNotEmpty()
  @IsUUID()
  intervention_id: string; 
}

File: cuaderno_mantenimiento_flutter/lib/infrastructure/dtos/create-intervention-details-dto.dart
class CreateInterventionDetailDto {
  final String elemento;
  final String estado;
  final String interventionId;
  final String? marca;

  CreateInterventionDetailDto({
    required this.elemento,
    required this.estado,
    required this.interventionId,
    this.marca,
  });

  Map<String, dynamic> toJson() {
    return {
      'elemento': elemento,
      'estado': estado,
      'intervention_id': interventionId,
      if (marca != null && marca!.isNotEmpty) 'marca': marca,
    };
  }
}
